weave/runtime.nim

```diff
@@ -29,7 +29,7 @@ else:
 # Runtime public routines
 # ----------------------------------------------------------------------------------
 
-proc init*(_: type Weave) =
+proc init*(_: type Weave, auxiliary: proc() {.gcsafe.} = nil) =
```

Comment:
The proc should be tagged

```
proc init*(_: type Weave, auxiliary: proc() {.nimcall, gcsafe.} | proc() {.nimcall, gcsafe.} = nil) =
```

so that nimcall and cdecl are allowed but not closures